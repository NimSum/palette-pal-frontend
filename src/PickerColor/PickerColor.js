import React, { Component } from 'react';

class PickerColor extends Component {
  constructor(props) {
    super(props);


  }
  
  render() {
    return (
      <div className="picker-color" style={{ backgroundColor: this.props.color, color: this.props.color }}>
        <p className="picker-color-value">{this.props.color}</p>
        <p className="picker-hold"><i className="far fa-square"></i>HOLD</p>
      </div>
    );
  }
}

export default PickerColor;