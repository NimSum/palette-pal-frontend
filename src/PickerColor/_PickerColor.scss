.picker-color {
  flex: 1;
  display: flex;
  flex-direction: column;
  justify-content: center;
  position: relative;
}

.picker-color-value {
  font-size: 20px;
  color: white;
  opacity: .5;
  text-transform: uppercase;
}

.picker-hold {
  font-size: 26px;
  color: white;
  // opacity: .5;
  font-weight: 600;
  position: absolute;
  bottom: 30px;
  left: 0;
  right: 0;
  margin-left: auto;
  margin-right: auto;
  cursor: pointer;
  opacity: 0;

  &:hover {
    opacity: .6;
  }

  &.hold {
    opacity: .6!important;
  }
}



// .hold, .color-value {
//   color: inherit;
//   // mix-blend-mode: difference;
//   // opacity: .7;
//   filter: brightness(25%);
// }