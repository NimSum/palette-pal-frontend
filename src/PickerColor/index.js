import React from 'react';

function PickerColor(props){
	const checkbox = props.held ? 'far fa-check-square' : 'far fa-square';
  const holdClass = props.held ? 'hold' : null;

	return (
		<div className="picker-color" style={{ backgroundColor: props.color }}>
      <p className="picker-color-value" style={{ color: props.getContrastColor(props.color) }}>{props.color}</p>
      <p className={`picker-hold ${holdClass}`} style={{ color: props.getContrastColor(props.color) }} onClick={() => props.toggleHold(props.id)}>
				<i className={checkbox} />HOLD
			</p>
		</div>
	);
}

export default PickerColor;
