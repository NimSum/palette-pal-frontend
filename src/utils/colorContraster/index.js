const colorContraster = color => {
  color = color.replace("#", "");
  var r = parseInt(color.substr(0, 2), 16);
  var g = parseInt(color.substr(2, 2), 16);
  var b = parseInt(color.substr(4, 2), 16);
  var yiq = ((r * 299) + (g * 587) + (b * 114)) / 1000;
  return (yiq >= 128) ? 'black' : 'white';
}

export default colorContraster;