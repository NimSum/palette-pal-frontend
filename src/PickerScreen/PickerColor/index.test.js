import React from 'react';
import PickerColor from './index';
import { shallow } from 'enzyme';
import colorContraster from '../../utils/colorContraster';
jest.mock('../../utils/colorContraster')

describe('PickerColor', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(< PickerColor />);
  })
  
  it('should match component PickerColor snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
})