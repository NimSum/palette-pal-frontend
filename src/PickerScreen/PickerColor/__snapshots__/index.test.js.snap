// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PickerColor should match component PickerColor snapshot 1`] = `ShallowWrapper {}`;
